src/pip/_internal/pep425tags.py

```diff
@@ -276,8 +293,16 @@ def get_supported(versions=None, noarch=False, platform=None,
             else:
                 # arch pattern didn't match (?!)
                 arches = [arch]
-        elif platform is None and is_manylinux1_compatible():
-            arches = [arch.replace('linux', 'manylinux1'), arch]
+        elif arch.startswith('manylinux2'):
+            # manylinux1 wheels run on manylinux2 systems.
```

Comment:
Given the removal of ncurses 5 from the ABI requirement, this implication isn't necessarily true: it's possible for a platform to be compatible with `manylinux2010` without being compatible with `manylinux`.

Reply:
In practice pip's heuristics for deciding whether a platform is compatible with manylinux1 and manylinux2010 don't actually do any checking of ncurses. I suppose it's possible for someone to provide a `_manylinux` module that disables `manylinux1` but enables `manylinux2010` (though this would be very unusual).

The branch below that just calls `is_manylinux2_compatible` and `is_manylinux1_compatible` seems like the simplest approach, and it's obviously correct. In fact I don't even understand how this branch could get hit in the first place...

Reply:
@njsmith The `_manylinux.is_manylinux1_compatible == False` override is exactly the case I had in mind. While that's unusual now, and while formally incompatible distros may choose not to set it on the basis of "ncurses dependencies are rare, while manylinux1 wheels are common", I think we want to start out with the position that both variants of Linux platform compatibility should be checked independently.

As for how this branch can be reached, it's only when `manylinux2010` is passed in specifically as the `platform` argument (presumably because the system checking the tags isn't actually a Linux system, or else isn't manylinux compatible).

So I think the best way to start out will be with a strict interpretation of the `manylinux2010` compatibility tag, and then treat "Opt-in to accepting manylinux1 wheels when explicitly specifying a manylinux2010 target" as a follow-on feature request.

Reply:
@wtolson pointed out that the PEP explicitly calls for `manylinux1` wheels to be treated as `manylinux2010` compatible, even with the `ncurses` dependency issue: https://www.python.org/dev/peps/pep-0571/#backwards-compatibility-with-manylinux1-wheels

I don't think it's worth going back and amending the PEP at this point, so let's just keep the branch, and add a reference to that part of the PEP in a comment.